addon/globalPlugins/webAccess/gui/properties.py

```diff
@@ -26,1011 +26,1031 @@
-        self.context = context
-        self.properties = properties
-        for props in self.properties:
-            if props.get_flag():
-                id = self.FindItem(props.get_displayName())
-                if id != -1:
-                    self.Delete(id)
-                    self.Bind(wx.EVT_MENU, props.updateFlag)
-                    return
-            else:
-                itemCheck = self.Append(wx.ID_ANY, self.displayLabel(props))
-                self.Bind(wx.EVT_MENU, props.updateFlag, itemCheck)
-
-    def displayLabel(self, props):
-        """
-        getAltFieldLabel func calls static function and filter the Editable
-        object and rule and return the appropriate display value
-        """
-        dataRule = self.context["data"]["rule"]
-        ruleType = dataRule.get("type")
-        if isinstance(props, EditableProperty):
-            return ListControl.getAltFieldLabel(ruleType, props.get_id())
-        else:
-            return props.get_displayName()
-
-    def ShowModal(self):
-        gui.mainFrame.prePopup(contextMenuName="Properties rules")
-        gui.mainFrame.PopupMenu(self)
-        gui.mainFrame.postPopup()
+	"""
+	Class displays the context menu for criteria panel properties
+	"""
+
+	def __init__(self, context, properties):
+		super(PropsMenu, self).__init__()
+		self.context = context
+		self.properties = properties
+		for props in self.properties:
+			if props.get_flag():
+				id = self.FindItem(props.get_displayName())
+				if id != -1:
+					self.Delete(id)
+					self.Bind(wx.EVT_MENU, props.updateFlag)
+					return
+			else:
+				itemCheck = self.Append(wx.ID_ANY, props.get_displayName())
+				self.Bind(wx.EVT_MENU, props.updateFlag, itemCheck)
+
+	def displayLabel(self, props):
+		"""
+		getAltFieldLabel func calls static function and filter the Editable
+		object and rule and return the appropriate display value
+		"""
+		dataRule = self.context["data"]["rule"]
+		ruleType = dataRule.get("type")
+		if isinstance(props, EditableProperty):
+			return ListControl.getAltFieldLabel(ruleType, props.get_id())
+		else:
+			return props.get_displayName()
+
+	def ShowModal(self):
+		gui.mainFrame.prePopup(contextMenuName="Properties rules")
+		gui.mainFrame.PopupMenu(self)
+		gui.mainFrame.postPopup()
 
 
 class PropertyType(Enum):
-    TOGGLE = "toggle"  # True / False
-    EDITABLE = "editable"
-    SINGLE_CHOICE = "singleChoice"
-    MULTIPLE_CHOICE = "multipleChoice"
+	TOGGLE = "toggle"  # True / False
+	EDITABLE = "editable"
+	SINGLE_CHOICE = "singleChoice"
+	MULTIPLE_CHOICE = "multipleChoice"
+
 
 class Property(ABC):
-    """
-    Abstract class for properties
-    """
-    @abstractmethod
-    def __init__(
-            self,
-            id: str,
-            display_name: str,
-            type: PropertyType
-    ):
-        self.__id = id
-        self.__display_name = display_name
-        self.__type = type
-
-    @abstractmethod
-    def get_id(self):
```

Comment:
May I ask why you are using an abstract getter here?
In general, is there a need for the use of getter/setter pairs all over this module?
If not, using direct attribute access is much more concise, and is the common _Python way_.
If getters/setters are indeed needed, you can use _properties_.